src/states/elder/mod.rs

```diff
@@ -677,6 +681,12 @@ impl Elder {
         }
     }
 
+    fn handle_section_info_ack(&mut self, sec_info: &SectionInfo, _src: XorName, dst: XorName) {
```

Comment:
This cannot be called when receiving a message as we are updating shared_state. (can only be done on consensus).
BTW, this is part of the other issue to handle it.

Reply:
What this mean is that function should not be in that PR, so is update_their_keys.

Reply:
Then in this PR, the `receive` can be totally skipped?

Reply:
Yes, i.e you don't have to do anything useful with the receive.
Looking at your test code, maybe you could replace your `sent_section_info_ack` with `received_section_info_ack`. and setting it to true is the only thing this PR does.

Reply:
hmm...   here we only `sent out section info ack`, and hasn't receive that yet.
so, I think it shall not be `received_section_info_ack` ?
or, you mean `received_section_info` ?

Reply:
I meant something like replace
```
            (AckMessage(sec_info), Section(src), Section(dst)) => {
                self.handle_section_info_ack(&sec_info, src, dst);
                Ok(())
            }
```

with 
```
            (AckMessage(sec_info), Section(src), Section(dst)) => {
                self.received_section_info_ack = true;
                Ok(())
            }
```

Reply:
Ah, ok